src/svm/svc.rs

```diff
@@ -102,6 +102,109 @@ pub struct SVCParameters<T: RealNumber, M: Matrix<T>, K: Kernel<T, M::RowVector>
     m: PhantomData<M>,
 }
 
+/// SVC grid search parameters
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone)]
+pub struct SVCSearchParameters<T: RealNumber, M: Matrix<T>, K: Kernel<T, M::RowVector>> {
+    /// Number of epochs.
+    pub epoch: Vec<usize>,
+    /// Regularization parameter.
+    pub c: Vec<T>,
+    /// Tolerance for stopping epoch.
+    pub tol: Vec<T>,
+    /// The kernel function.
+    pub kernel: Vec<K>,
```

Comment:
This won't work, since we may want to compare multi different types of kernels, and Vec must contain only a single type. SVM Kernel is the only example of this issue among the existing parameters. 

The options to fix don't appear straightforward to me. I haven't had success boxing the kernels, since they are templated themselves, and can't be `dyn`. I started looking at a potential `enum` based solution where we could store equal sized enum values that represent the kernels in a vec, but haven't gotten it working yet. 

I'm adding a test case to illustrate the issue.

Reply:
```rust
enum Kernels<T> {
  Linear,
  Polynomial(T, T, T),
}

fn main() {
  let v = vec![Kernels::<f32>::Linear, Kernels::<f32>::Polynomial(32.0, 1.0, 5.0)];
}
```

Reply:
I've started outlining a solution with the commit below that moves from the current `Kernel` implementations to an Enum version, but I believe this would be a breaking change.